### PR TITLE
pixart-rf: Only show one PixArt receiver device per physical device

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
@sam412081go

File: plugins/pixart-rf/fu-pxi-receiver-device.c [hunk 1]
```diff
@@ -893,6 +893,23 @@ fu_pxi_receiver_device_setup(FuDevice *device, GError **error)
...
+	usb_parent = fu_udev_device_get_parent_with_subsystem(FU_UDEV_DEVICE(device), "usb", error);
+	if (usb_parent == NULL)
+		return FALSE;
+	if (!fu_udev_device_get_sysfs_attr_uint64(usb_parent, "bInterfaceNumber", &iface_nr, error))
+		return FALSE;
+	if (iface_nr != 0) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_NOT_SUPPORTED,
+			    "only USB interface 0 supported");
```
Comment (superm1) [inline plugins/pixart-rf/fu-pxi-receiver-device.c:909]: if this isn't always true it's better to put the interface number in a quirk

**Comment by sam412081go** [comment]
I pulled this commit for testing, and currently, get-devices only shows one device:
Screenshot from 2024-06-18 10-51-51


But result of emulation-save still same as https://github.com/fwupd/fwupd/pull/7330
[daemon_log.txt](https://github.com/user-attachments/files/15879526/daemon_log.txt)
[fwupdmgr_log.txt](https://github.com/user-attachments/files/15879525/fwupdmgr_log.txt)

**Comment by hughsie** [comment]
> But result of emulation-save still same as https://github.com/fwupd/fwupd/pull/7330

Yes, and I worked out why -- I gave some really bad advice -- and probably need to apologize for wasting your time! I thought pixart-rf was using a baseclass of FuUsbDevice, but it's using hidraw, so it's using FuUdevDevice. This means the emulation through libgusb isn't going to work. I need to do a few things to make it work:

 * Move `GUsbDeviceEvent` up as `FuDeviceEvent`
 * Move plugins from `g_usb_device_control_transfer` to `fu_usb_device_control_transfer` and save the events there
 * Move the `->load()` and `->save()` action to FuBackend rather than GUsbContext
 
Maybe that also allows us to drop libgusb as a dep (using libusb directly), which gets us closer to our Android port too. I'll use your 2862 devboard as the test for this new `FuDeviceEvent` future.

**Comment by hughsie** [comment]
> should I include your modificatoin of this commit in the my PR together?

A different PR please; I'll merge this one now.

**Pull request merged**